File: examples/typescript/create-plan.ts
#!/usr/bin/env npx tsx
/**
 * Create a new plan with GUID.
 * Usage: npx tsx create-plan.ts "Plan Name" repo-name
 */
import { randomUUID } from "crypto";
import { mkdirSync, writeFileSync } from "fs";
import { dirname } from "path";

interface PlanOptions {
  name: string;
  repo: string;
}

function createPlan({ name, repo }: PlanOptions): string {
  // Short GUID: 5 chars from uuid
  const guid = randomUUID().replace(/-/g, "").slice(0, 5);
  // Plan ID: repo prefix + guid
  const planId = `${repo.slice(0, 2)}${guid}`;

  const content = `---
id: ${planId}
status: draft
repo: ${repo}
created: ${new Date().toISOString()}
---

# ${name}

## Goal

## Tasks
- [ ]
`;

  const slug = name.toLowerCase().replace(/\s+/g, "-");
  const path = `plans/${repo}/${planId}-${slug}.md`;

  // Create directory and write file
  mkdirSync(dirname(path), { recursive: true });
  writeFileSync(path, content);

  return path;
}

// CLI
const [, , name, repo] = process.argv;

if (!name || !repo) {
  console.log("Usage: npx tsx create-plan.ts 'Plan Name' repo-name");
  process.exit(1);
}

const path = createPlan({ name, repo });
console.log(`Created: ${path}`);

File: examples/python/create_plan.py
#!/usr/bin/env python3
"""Create a new plan with GUID.

Usage: ./create_plan.py "Plan Name" repo-name
"""
import sys
import uuid
from pathlib import Path
from datetime import datetime


def create_plan(name: str, repo: str) -> Path:
    """Generate a plan file with unique ID."""
    # Short GUID: 5 chars from uuid4
    guid = uuid.uuid4().hex[:5]
    # Plan ID: repo prefix + guid (e.g., "wf001a")
    plan_id = f"{repo[:2]}{guid}"

    # Frontmatter + template
    content = f"""---
id: {plan_id}
status: draft
repo: {repo}
created: {datetime.now().isoformat()}
---

# {name}

## Goal

## Tasks
- [ ]
"""

    # Create plans directory structure
    slug = name.lower().replace(" ", "-")
    path = Path(f"plans/{repo}/{plan_id}-{slug}.md")
    path.parent.mkdir(parents=True, exist_ok=True)
    path.write_text(content)
    return path


if __name__ == "__main__":
    if len(sys.argv) < 3:
        print("Usage: ./create_plan.py 'Plan Name' repo-name")
        sys.exit(1)

    name = sys.argv[1]
    repo = sys.argv[2]
    path = create_plan(name, repo)
    print(f"Created: {path}")
